### DISCUSSION TITLE
Desktop Icons

### DISCUSSION THREAD

**Discussion by rijenkii** [WayfireWM/wayfire#1711] [Q&A] [answered]
I want to preface this by stating that I know little about developing X applications, even less about Wayland apps, and even less about Wayland compositors.

If I want to add desktop icons to Wayfire (be it contents of the regular old "Desktop" folder, or a list of hidden windows a-la Win 3.11), in what direction should I start my research?

IIRC in X icons were drawn by regular applications (xfdesktop or nautilus).
Should I try to implement desktop icons the same way as in X (some kind of regular Wayland app), or as a plugin/extension to Wayfire?

**Comment by wb9688** [comment, marked as answer]
That could be implemented as a client that uses the layer shell protocol.

Fwiw I have made a PR to PCManFM to support that, see https://github.com/lxde/pcmanfm/pull/4 but you would need to compile it yourself. Note that it would also need my PRs to LibFM like https://github.com/lxde/libfm/pull/70 to function properly.

**Comment by stefonarch** [comment]
With the [lxqt-desktop-shell plugin ](https://gitlab.com/wayfireplugins/lxqt-desktop-shell/) the desktop from PCmanFM-qt is working perfectly.
